pyomo/contrib/solver/solvers/ipopt.py

```diff
@@ -380,23 +393,27 @@ def solve(self, model, **kwds):
                     timeout = None
 
                 ostreams = [io.StringIO()] + config.tee
-                with TeeStream(*ostreams) as t:
-                    timer.start('subprocess')
-                    process = subprocess.run(
-                        cmd,
-                        timeout=timeout,
-                        env=env,
-                        universal_newlines=True,
-                        stdout=t.STDOUT,
-                        stderr=t.STDERR,
-                        check=False,
-                    )
-                    timer.stop('subprocess')
-                    # This is the stuff we need to parse to get the iterations
-                    # and time
-                    (iters, ipopt_time_nofunc, ipopt_time_func, ipopt_total_time) = (
-                        self._parse_ipopt_output(ostreams[0])
-                    )
+                try:
+                    with TeeStream(*ostreams) as t:
+                        timer.start('subprocess')
+                        process = subprocess.run(
+                            cmd,
+                            timeout=timeout,
+                            env=env,
+                            universal_newlines=True,
+                            stdout=t.STDOUT,
+                            stderr=t.STDERR,
+                            check=False,
+                        )
+                        timer.stop('subprocess')
+                except OSError:
+                    err = sys.exc_info()[1]
+                    msg = 'Could not execute the command: %s\tError message: %s'
+                    raise ApplicationError(msg % (cmd, err))
+
+                # This is the stuff we need to parse to get the iterations
+                # and time
+                parsed_output_data = self._parse_ipopt_output(ostreams[0])
```

Comment:
I think we should have an option to turn off this parsing.

Reply:
Thinking out loud:

- For what reason? We have to parse anyway to get timing info, which we have agreed is the new universal standard for solver results.
- If we do want to only parse for timing and not for other stuff, what is the default (e.g., is the default `True` - parse or `False` - don't parse)?
- In what instance is it not helpful to have this information?